packages/client/src/v2-events/features/events/actions/declare/DeclareActionMenu.tsx

```diff
@@ -0,0 +1,344 @@
+   * We do this by 'looking in to the future' by applying the would-be actions to the event,
+   * and checking if the validate and register actions are still allowed.
+   */
+  function isDirectActionPossible(
+    actionType: typeof ActionType.VALIDATE | typeof ActionType.REGISTER
+  ) {
+    if (!userDetails) {
+      return false
+    }
+
+    const eventAfterDeclare = {
+      ...event,
+      actions: event.actions.concat({
+        type: ActionType.DECLARE,
+        id: 'placeholder' as UUID,
+        transactionId: 'placeholder' as UUID,
+        createdByUserType: TokenUserType.enum.user,
+        createdByRole: userDetails.role.id,
+        declaration,
+        annotation,
+        createdAt: new Date().toISOString(),
+        createdBy: userDetails.id,
+        originalActionId: null,
+        status: 'Accepted',
+        createdBySignature: undefined,
+        createdAtLocation: userDetails.primaryOffice.id as UUID
+      })
+    }
+
+    const eventIndexAfterDeclare = getCurrentEventState(
+      eventAfterDeclare,
+      eventConfiguration
+    )
+
+    const validateActionConfig = getActionConfig({
+      eventConfiguration,
+      actionType: ActionType.VALIDATE
+    })
+
+    if (!validateActionConfig) {
+      return false
+    }
+
+    const validateIsAvailable = isActionAvailable(
+      validateActionConfig,
+      eventIndexAfterDeclare,
+      validatorContext
+    )
+
+    if (actionType === ActionType.VALIDATE) {
+      return validateIsAvailable
+    }
+
+    const eventAfterValidate = {
+      ...eventAfterDeclare,
+      actions: eventAfterDeclare.actions.concat({
+        type: ActionType.VALIDATE,
+        id: 'placeholder' as UUID,
+        transactionId: 'placeholder' as UUID,
+        createdByUserType: TokenUserType.enum.user,
+        createdByRole: userDetails.role.id,
+        declaration,
+        annotation,
+        createdAt: new Date().toISOString(),
+        createdBy: userDetails.id,
+        originalActionId: null,
+        status: 'Accepted',
+        createdBySignature: undefined,
+        createdAtLocation: userDetails.primaryOffice.id as UUID
+      })
+    }
+
+    const registerActionConfig = getActionConfig({
+      eventConfiguration,
+      actionType: ActionType.REGISTER
+    })
+
+    if (!registerActionConfig) {
+      return false
+    }
+
+    const eventIndexAfterValidate = getCurrentEventState(
+      eventAfterValidate,
+      eventConfiguration
+    )
+
+    return isActionAvailable(
+      registerActionConfig,
+      eventIndexAfterValidate,
+      validatorContext
+    )
+  }
+
+  return {
+    modals: [modal, rejectionModal, saveAndExitModal, deleteDeclarationModal],
+    actions: [
+      {
+        icon: 'Check' as const,
+        label: actionLabels[ActionType.REGISTER],
+        onClick: async () => handleDeclaration(ActionType.REGISTER),
+        hidden: !isActionAllowed(ActionType.REGISTER),
+        disabled: incomplete || !isDirectActionPossible(ActionType.REGISTER)
```

Comment:
shouldn't the buttons be disabled when the form has some errors? I see we didn't have them previously. If it proves out to be too much additional work, we can create a small ticket for it as well